src/android/com/microsoftopentechnologies/intellij/wizards/createvm/CreateStorageAccountForm.java

```diff
@@ -0,0 +1,250 @@
+                return super.getListCellRendererComponent(jList, ((ReplicationTypes) o).getDescription(), i, b, b1);
+            }
+        });
+    }
+
+
+    private void validateEmptyFields() {
+        boolean allFieldsCompleted = !(
+                nameTextField.getText().isEmpty() || regionOrAffinityGroupComboBox.getSelectedObjects().length == 0);
+
+        buttonOK.setEnabled(allFieldsCompleted);
+    }
+
+    private void onOK() {
+
+        if (nameTextField.getText().length() < 3
+                || nameTextField.getText().length() > 24
+                || !nameTextField.getText().matches("[a-z0-9]+")) {
+            JOptionPane.showMessageDialog(this, "Invalid storage account name. The name should be between 3 and 24 characters long and \n" +
+                    "can contain only lowercase letters and numbers.", "Error creating the storage account", JOptionPane.ERROR_MESSAGE);
+            return;
+        }
+
+        setCursor(Cursor.getPredefinedCursor(Cursor.WAIT_CURSOR));
+        createProgressBar.setVisible(true);
+
+        try {
+            String name = nameTextField.getText();
+            String region = (regionOrAffinityGroupComboBox.getSelectedItem() instanceof Location) ? regionOrAffinityGroupComboBox.getSelectedItem().toString() : "";
+            String affinityGroup = (regionOrAffinityGroupComboBox.getSelectedItem() instanceof AffinityGroup) ? regionOrAffinityGroupComboBox.getSelectedItem().toString() : "";
+            String replication = replicationComboBox.getSelectedItem().toString();
+
+            storageAccount = new StorageAccount(name, replication, region, affinityGroup, "", subscription.getId().toString());
+            AzureSDKManagerImpl.getManager().createStorageAccount(storageAccount);
+
+            onCreate.run();
+        } catch (AzureCmdException e) {
+            UIHelper.showException("Error creating cloud service", e);
+        }
+
+        setCursor(Cursor.getDefaultCursor());
+
+        this.setVisible(false);
+        dispose();
+    }
+
+    private void onCancel() {
+        dispose();
+    }
+
+    public void fillFields(final Subscription subscription, Project project) {
+        this.subscription = subscription;
+
+        subscriptionComboBox.addItem(subscription.getName());
+
+        regionOrAffinityGroupComboBox.addItem("<Loading...>");
+
+        ProgressManager.getInstance().run(new Task.Backgroundable(project, "Loading regions...", false) {
+            @Override
+            public void run(ProgressIndicator progressIndicator) {
+
+                progressIndicator.setIndeterminate(true);
+
+                try {
+                    final java.util.List<AffinityGroup> affinityGroups = AzureSDKManagerImpl.getManager().getAffinityGroups(subscription.getId().toString());
+                    final java.util.List<Location> locations = AzureSDKManagerImpl.getManager().getLocations(subscription.getId().toString());
```

Comment:
As mentioned before, this can be run in parallel. Also, we seem to make the same two calls from the `CreateCloudServiceForm` as well. It might be worthwhile to build a cache layer of some kind in the `AzureSDKManagerImpl` to avoid repeated calls to RDFE only to fetch exactly the same data.